engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/Atom.java

```diff
@@ -95,9 +87,15 @@ private String toString(boolean shouldParen) {
    * @return a textual representation of this Atom.
    */
   @Override
-  @CompilerDirectives.TruffleBoundary
   public String toString() {
-    return toString(false);
+    return toString(10);
+  }
+
+  @CompilerDirectives.TruffleBoundary
+  private String toString(int depth) {
+    StringBuilder sb = new StringBuilder();
```

Comment:
Reuses single instance of `StringBuilder` for concatenation of all (nested) `Atom` - more effective than allocating new `StringBuilder` for each `Atom`.